realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -312,6 +365,38 @@ public void onError(Throwable error) {
         });
     }
 
+    @Test
+    public void executeTransactionAsync_onSuccessOnNonLooperThreadThrows() {
+        Realm realm = Realm.getInstance(configFactory.createConfiguration());
+        thrown.expect(IllegalStateException.class);
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+
+            }
+        }, new Realm.Transaction.OnSuccess() {
+            @Override
+            public void onSuccess() {
+            }
+        });
+    }
+
+    @Test
+    public void executeTransactionAsync_onErrorOnNonLooperThreadThrows() {
+        Realm realm = Realm.getInstance(configFactory.createConfiguration());
+        thrown.expect(IllegalStateException.class);
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+
+            }
+        }, new Realm.Transaction.OnError() {
```

Comment:
The test is `executeTransactionAsync_onErrorOnNonLooperThreadThrows`. it is testing if the transaction has an error block, then it should throw if it is called on a looper thread because of the onError cannot be delivered.